### PR TITLE
chore: GraphQL::ObjectType was deprecated and removed

### PR DESCRIPTION
**Pull request by voidfiles**
The GraphQL::ObjectType was deprecated and removed in graphql 2.0.0.
When this project was upgraded to 2.0.0 the code failed because it
was checking if types were GraphQL::ObjectType.

By checking if GraphQL::ObjectType is defined before checking if the
type is_a GraphQL::ObjectType the library should be compatible across
multiple upstream versions.

### CODE REVIEW

**Review by daemonsy — COMMENTED** [review]
This looks good to me. Can you rebase this with latest `main`, get the Gemfiles etc up to date?

FYI, we might consider dropping support for GraphQL < 1.12 soon which means we can eventually remove checking the types.

File: Gemfile [hunk 1]
```diff
@@ -4,4 +4,4 @@ source 'https://rubygems.org'
 
 gemspec
 
-gem 'graphql', '1.10.14'
+gem 'graphql'
```
[the remark this answers is not in the document]
Comment (voidfiles) [reply Gemfile:7]: I think it was just ignorance. I will put it back.

**Comment by voidfiles** [comment]
Yea, I will make some tweaks, and rebase.

**Review by daemonsy — APPROVED** [review]
LGTM :+1:

**Pull request merged**